error/s2n_errno.c

```diff
@@ -161,6 +161,7 @@ static const char *no_such_error = "Internal s2n error";
     ERR_ENTRY(S2N_ERR_FREE_STATIC_BLOB, "Cannot free a static blob") \
     ERR_ENTRY(S2N_ERR_RESIZE_STATIC_BLOB, "Cannot resize a static blob") \
     ERR_ENTRY(S2N_ERR_NO_AVAILABLE_BORINGSSL_API, "BoringSSL does not support this API") \
+    ERR_ENTRY(S2N_ERR_NO_AVAILABLE_BSSL_AWSLC_API, "BoringSSL/AWS-LC does not support this API") \
```

Comment:
Update error message to `S2N_ERR_NO_SUPPORTED_LIBCRYPTO_API` to reflect changes discussed here: https://github.com/awslabs/s2n/pull/2273/files#r483487342